File: Data/Python/stack.py
class Stack:
    """
    Stack Data Structure (LIFO):
    - Supports push (add to top) and pop (remove from top).
    - Uses Python list internally.
    - Time Complexity: O(1) for push/pop
    """
    def __init__(self):
        self.stack = []

    def push(self, value):
        self.stack.append(value)  # Add element to top

    def pop(self):
        if self.is_empty():
            return None  # Return None if stack is empty
        return self.stack.pop()  # Remove top element

    def peek(self):
        if self.is_empty():
            return None
        return self.stack[-1]  # Return top element without removing

    def is_empty(self):
        return len(self.stack) == 0

# ✅ Example
s = Stack()
s.push(10)
s.push(20)
print("Top:", s.peek())  # 20
print("Popped:", s.pop())  # 20
print("Top After Pop:", s.peek())  # 10

File: C++/stack.cpp
#include <iostream>
#include <stack>
using namespace std;

// ✅ Use-case of built-in stack
int main() {
    stack<int> st;

    // Push elements onto stack
    st.push(10);  // Stack: [10]
    st.push(20);  // Stack: [10, 20]
    st.push(30);  // Stack: [10, 20, 30]

    cout << "Top element: " << st.top() << endl;  // Output: 30

    st.pop();  // Remove 30
    cout << "After pop, top: " << st.top() << endl;  // Output: 20
}
